.github/workflows/docker.yml

```diff
@@ -13,58 +15,107 @@ concurrency:
+      attestations: write
+      packages: write
+      id-token: write
 
     steps:
+      - name: Set PUSH_PACKAGE due to schedule
+        if: github.event.schedule=='*/0 * * * *'
+        run: |
+          echo "PUSH_PACKAGES=true" >> $GITHUB_ENV
       - name: Checkout
         uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 # v5.0.0
-
+      - name: Set up Docker
+        uses: docker/setup-docker-action@v4
+        with:
+          daemon-config: |
+            {
+              "debug": false,
+              "features": {
+                "containerd-snapshotter": true
+              }
+            }
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Extract container metadata
+        id: meta
+        uses: docker/metadata-action@v5
+        with:
+          images: |
+            ghcr.io/osgeo/gdal
+          tags: |
+            type=ref,event=branch
+            type=ref,event=pr
+            type=semver,pattern={{version}}
+            type=semver,pattern={{major}}.{{minor}}
+        env:
+          DOCKER_METADATA_ANNOTATIONS_LEVELS: manifest,index
       - name: Build
         shell: bash -l {0}
         run: |
-          docker run --rm --privileged linuxkit/binfmt:v0.8
-          cd docker/${{ matrix.image_name }}
-          if test "${{ matrix.name }}" = "ubuntu-full-amd64-proprietary-sdks"; then
-            ./build.sh --platform linux/${{ matrix.arch }} --with-oracle --with-mrsid --with-ecw
+          cd docker/${{ matrix.base }}
+          if test "${{ matrix.proprietary_sdks }}" = "true"; then
+          echo "PUSH_PACKAGES=false" >> $GITHUB_ENV
+            ./build.sh --platform linux/${{ matrix.platform }} --with-multi-arch --with-oracle --with-mrsid --with-ecw
           else
-            ./build.sh --platform linux/${{ matrix.arch }}
+            ./build.sh --platform linux/${{ matrix.platform }} --with-multi-arch
           fi
-          docker run --rm ghcr.io/osgeo/gdal:${{ matrix.image_name }}-latest-${{ matrix.arch }} gdalinfo --formats
-          docker run --rm ghcr.io/osgeo/gdal:${{ matrix.image_name }}-latest-${{ matrix.arch }} ogrinfo --formats
+          docker run --rm ghcr.io/osgeo/gdal:${{ matrix.base }}-latest-${{ matrix.platform }} gdalinfo --formats
+          docker run --rm ghcr.io/osgeo/gdal:${{ matrix.base }}-latest-${{ matrix.platform }} ogrinfo --formats
+
+  create-manifest:
+    permissions:
+      contents: read
+      packages: write
+      attestations: write
+      id-token: write
+    runs-on: ubuntu-latest
+    needs: docker_builds
+    strategy:
+      fail-fast: false
+      matrix:
+        tag: ["ubuntu-full", "ubuntu-small", "alpine-small", "alpine-normal"]
+    steps:
+      - name: Set PUSH_PACKAGE due to schedule
+        if: github.event.schedule=='*/0 * * * *'
```

Comment:
I don't know if environment variables are transferred across jobs, but if they are transferred this will overwrite the `PUSH_PACKAGE` from the build step in the previous job.

Reply:
https://docs.github.com/en/actions/how-tos/write-workflows/choose-what-workflows-do/use-variables#passing-values-between-steps-and-jobs-in-a-workflow

> If you want to pass a value from a step in one job in a workflow to a step in another job in the workflow, you can define the value as a job output. You can then reference this job output from a step in another job. For more information, see [Workflow syntax for GitHub Actions](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idoutputs).

Extra steps need to be made to pass an environment variable that was set in the step of one job to another job.